packages/services/f-analytics/src/services/_tests/analytics.service.test.js

```diff
@@ -0,0 +1,118 @@
+import AnalyticService from '@/services/analytics.service';
+import {
+    UPDATE_PLATFORM_DATA,
+    UPDATE_EVENTS,
+    CLEAR_EVENTS
+} from '@/store/mutation-types';
+import {
+    defaultState,
+    newEvent,
+    options
+} from '@/tests/helpers/setup';
+
+describe('Analytic Service ::', () => {
+    let store;
+    let storeDispatchSpy;
+    let service;
+    let req;
+    let windowsPushSpy;
+    let state;
+
+    beforeEach(() => {
+        // Arrange - store
+        storeDispatchSpy = jest.fn();
+        state = [`${options.namespace}`];
+        state[`${options.namespace}`] = { ...defaultState };
+        store = {
+            state,
+            dispatch: storeDispatchSpy
+        };
+        // Arrange - request
+        req = {
+            headers: jest.fn(() => ['user-agent'])
+        };
+        // Arrange - window state
+        jest.spyOn(window.navigator, 'userAgent', 'get').mockReturnValue('test-agent-string');
+        windowsPushSpy = jest.fn();
+        const originalWindow = { ...window };
+        jest.spyOn(global, 'window', 'get').mockImplementation(() => ({
+            ...originalWindow,
+            dataLayer: {
+                push: windowsPushSpy
+            }
+        }));
+        service = new AnalyticService(store, req, options);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should instantiate a new instance', () => {
+        // Act
+        const instance = () => new AnalyticService(store, req, options);
+
+        // Assert
+        expect(instance).not.toThrowError();
+        expect(instance).not.toBeNull();
```

Comment:
I could be wrong here but would `.toBeDefined()` be better here since you could get an undefined result which would pass here?